src/query/api/v1/handler/prometheus/native/read_test.go

```diff
@@ -116,29 +117,39 @@ func newReadRequest(t *testing.T, params url.Values) *http.Request {
 }
 
 type testSetup struct {
-	Storage mock.Storage
-	Handler *PromReadHandler
+	Storage     mock.Storage
+	Handlers    testSetupHandlers
+	TimeoutOpts *prometheus.TimeoutOpts
+}
+
+type testSetupHandlers struct {
+	Read        *PromReadHandler
+	InstantRead *PromReadInstantHandler
 }
 
 func newTestSetup() *testSetup {
 	mockStorage := mock.NewMockStorage()
-
+	scope := tally.NoopScope
+	engine := executor.NewEngine(mockStorage, scope,
+		time.Minute, nil)
+	tagOpts := models.NewTagOptions()
+	limitsConfig := &config.LimitsConfiguration{}
+	keepNans := false
 	return &testSetup{
 		Storage: mockStorage,
-		Handler: NewPromReadHandler(
-			executor.NewEngine(mockStorage, tally.NewTestScope("test", nil), time.Minute, nil),
-			models.NewTagOptions(),
-			&config.LimitsConfiguration{},
-			tally.NewTestScope("", nil),
-			timeoutOpts,
-			false,
-		),
+		Handlers: testSetupHandlers{
+			Read: NewPromReadHandler(engine, tagOpts, limitsConfig,
```

Comment:
nit: think it might read a little nicer if these handlers were defined before the return statement; line breaks would probably look a bit more readable

Reply:
Sg.